libexecutive/Executive.cpp

```diff
@@ -48,42 +47,42 @@ void Executive::accrueSubState(SubState& _parentContext)
         _parentContext += m_ext->sub();
 }
 
-void Executive::initialize(Transaction const& _transaction)
+void Executive::initialize(Transaction::Ptr _transaction)
 {
     m_t = _transaction;
-    m_baseGasRequired = m_t.baseGasRequired(g_BCOSConfig.evmSchedule());
+    m_baseGasRequired = m_t->baseGasRequired(g_BCOSConfig.evmSchedule());
 
     verifyTransaction(ImportRequirements::Everything, m_t, m_envInfo.header(), m_envInfo.gasUsed());
 
-    if (!m_t.hasZeroSignature())
+    if (!m_t->hasZeroSignature())
     {
         // No need nonce increasing sequently at all. See random id for more.
 
         // Avoid unaffordable transactions.
-        bigint gasCost = (bigint)m_t.gas() * m_t.gasPrice();
-        bigint totalCost = m_t.value() + gasCost;
+        bigint gasCost = (bigint)m_t->gas() * m_t->gasPrice();
+        bigint totalCost = m_t->value() + gasCost;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 'totalCost' is assigned a value that is never used.](https://app.codacy.com/app/fisco/FISCO-BCOS/pullRequest?prid=4731778)